# HG changeset patch
# User Stuart Rackham <srackham@methods.co.nz>
# Date 1228260423 -46800
# Node ID 2f2256c31fbe542bdd2bbb2753989f2444127238
# Parent  baf9c1979fb15ccd412e3ef5f8387493e2640489
FIXED: Excluding double-quotes from unquoted attribute values resulted in backward incompatibility, double-quotes reenbled.

diff -r baf9c1979fb1 -r 2f2256c31fbe asciidoc.py
--- a/asciidoc.py	Tue Dec 02 22:37:34 2008 +1300
+++ b/asciidoc.py	Wed Dec 03 12:27:03 2008 +1300
@@ -324,11 +324,7 @@ def parse_attributes(attrs,dict):
             if not (isinstance(v,str) or isinstance(v,int) or isinstance(v,float) or v is None):
                 raise
     except:
-        # Non-quoted attributes cannot contain double-quote characters, this
-        # restriction catches most quoted attribute syntax errors.
-        if s.find('"') >= 0:
-            error('malformed attributes: %s' % s)
-            return
+        s = s.replace('"','\\"')
         s = s.split(',')
         s = map(lambda x: '"' + x.strip() + '"', s)
         s = ','.join(s)
diff -r baf9c1979fb1 -r 2f2256c31fbe doc/asciidoc.txt
--- a/doc/asciidoc.txt	Tue Dec 02 22:37:34 2008 +1300
+++ b/doc/asciidoc.txt	Wed Dec 03 12:27:03 2008 +1300
@@ -3464,9 +3464,8 @@ to pass parameters to macros, blocks and
 - The list consists of zero or more positional attribute values
   followed by zero or more named attribute values.
 - Attribute values are enclosed in quotation mark (") characters.
-- If the attribute list contains no named attributes and the attribute
-  values contain no commas or quotation marks then quoting is
-  unnecessary.
+- If the attribute list only contains positional attribute values and
+  the values contain no commas then quoting is unnecessary.
 
 Here are three examples (a single unquoted positional attribute; three
 unquoted attribute values; one positional attribute followed by two
@@ -3479,11 +3478,10 @@ named attributes):
 .Attribute list behavior
 - If one or more attribute values contains a comma the all values must
   be quoted (enclosed in quotation characters).
-- If the list contains any named or quoted attributes the all string
+- If the list contains any named or quoted attributes then all string
   attribute values must be quoted.
-- To include a quotation mark (") character in an attribute value the
-  value must be quoted and the quotation mark escaped with a
-  backslash.
+- To include a quotation mark (") character in a quoted attribute
+  value the the quotation mark must be escaped with a backslash.
 - List attributes take precedence over existing attributes.
 - List attributes can only be referenced in configuration file markup
   templates and tags, they are not available inside the document.
